drivers/sensor/hdc20x0/hdc20x0.c

```diff
@@ -0,0 +1,243 @@
+		LOG_ERR("Failed to get humidity MSB ");
+		return -EIO;
+	}
+	data->humidity_val2 = val;
+	LOG_DBG("humidity , val1:0x%x, val2:0x%x", data->humidity_val1,
+		data->humidity_val2);
+	/*https://www.ti.com/lit/ds/symlink/hdc20x0.pdf*/
+	/*7.6.4 Address 0x03 Humidity MSB*/
+	rh = ((data->humidity_val2 * 256 + data->humidity_val1) / 65536.0) *
+	     100;
+
+	data->humidity_val1 = (int)rh;
+	data->humidity_val2 = (rh - data->humidity_val1) * 1000000LL;
+	return 0;
+}
+
+static int hdc20x0_read_all(struct hdc20x0_data *data,
+			    const struct hdc20x0_cfg *config)
+{
+	if (hdc20x0_read_temperature(data, config)) {
+		return -EIO;
+	}
+	if (hdc20x0_read_humidity(data, config)) {
+		return -EIO;
+	}
+	return 0;
+}
+
+static int hdc20x0_init(const struct device *dev)
+{
+	struct hdc20x0_data *data = dev->data;
+	const struct hdc20x0_cfg *config = dev->config;
+	int ret;
+
+	data->i2c_master = device_get_binding(config->i2c_bus_name);
+	if (data->i2c_master == NULL) {
+		LOG_ERR("Failed to get I2C device");
+		return -EINVAL;
+	}
+
+	LOG_DBG("Init hdc20x0");
+
+	/* Enable Automatic Measurement Mode at 5Hz */
+	if (i2c_reg_write_byte(data->i2c_master, config->i2c_addr,
+			       hdc20x0_REG_RESET_DRDY_INT_CONF, hdc20x0_AMM)) {
+		return -EIO;
+	}
+	/*
+	 * We enable both temp and humidity measurement.
+	 * However the measurement won't start even in AMM until triggered.
+	 */
+	ret = i2c_reg_write_byte(data->i2c_master, config->i2c_addr,
+				 hdc20x0_REG_CONF, hdc20x0_MEAS_TRIG);
+	if (ret) {
+		LOG_ERR("Unable to set up measurement\n");
+		return ret;
+	}
+
+	LOG_DBG("Init OK");
+	return 0;
+}
+
+static int hdc20x0_sample_fetch(const struct device *dev,
+				enum sensor_channel chan)
+{
+	struct hdc20x0_data *data = dev->data;
+	const struct hdc20x0_cfg *config = dev->config;
+
+	LOG_DBG("Fetching sample from hdc20x0");
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		if (hdc20x0_read_temperature(data, config)) {
+			LOG_ERR("Failed to measure temperature");
+			return -EIO;
+		}
+		break;
+	case SENSOR_CHAN_HUMIDITY:
+		if (hdc20x0_read_humidity(data, config)) {
+			LOG_ERR("Failed to measure humidity");
+			return -EIO;
+		}
+		break;
+	case SENSOR_CHAN_ALL:
+		if (hdc20x0_read_all(data, config)) {
+			LOG_ERR("Failed to measure temperature and humidity");
+			return -EIO;
+		}
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int hdc20x0_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct hdc20x0_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		val->val1 = data->temp_val1;
+		val->val2 = data->temp_val2;
+		break;
+	case SENSOR_CHAN_HUMIDITY:
+		val->val1 = data->humidity_val1;
+		val->val2 = data->humidity_val2;
+		break;
+	case SENSOR_CHAN_ALL:
+		val->val1 = data->temp_val1;
+		val->val2 = data->temp_val2;
+		(val + 1)->val1 = data->humidity_val1;
+		(val + 1)->val2 = data->humidity_val2;
```

Comment:
I think it is a matter of implementation prereference. May I suggest the following? Not sure if you like it:

val[0].val1 = data->temp_val1;
val[0].val2 = ...
val[1].val1 = ...
val[1].val2=...